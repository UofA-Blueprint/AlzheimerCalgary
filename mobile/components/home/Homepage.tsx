import { Text, View } from 'react-native';
import { useState } from 'react';
import StaffHome from './StaffHome';
import CaregiverHome from './CaregiverHome';
import ClientHome  from './ClientHome';

interface RouterProps {
  role: string
}

const RoleRouter = ({role}: RouterProps) => {
    switch (role) {
        case 'staff':
            return <StaffHome />;
            break;
        case 'caregiver':
            return <CaregiverHome />;
            break;
        case 'client':
            return <ClientHome />;
            break;
        default:
            return (
                <Text>No role</Text>
            );
    };
};

const Homepage = () => {
    const [userRole, setUserRole] = useState('caregiver');

    return (
        <View>
            <Text>Homepage</Text>
            <Text>Your role: {userRole}</Text>
            <RoleRouter role={userRole}/>
        </View>
    )
};

export default Homepage;