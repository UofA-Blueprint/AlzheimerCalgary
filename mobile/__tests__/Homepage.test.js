import React from 'react';
import renderer from 'react-test-renderer';
import Homepage from '../components/home/Homepage'

test('renders correctly', () => {
  const tree = renderer.create(<Homepage />).toJSON();
  expect(tree).toMatchSnapshot();
});