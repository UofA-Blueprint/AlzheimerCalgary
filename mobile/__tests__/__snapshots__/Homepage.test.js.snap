// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text>
    Homepage
  </Text>
  <Text>
    Your role: 
    caregiver
  </Text>
  <View>
    <Text>
      Caregiver Homepage
    </Text>
  </View>
</View>
`;
